src/test/java/teammates/test/cases/datatransfer/FeedbackRubricQuestionDetailsTest.java

```diff
@@ -0,0 +1,319 @@
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-0-0", new String[] { "1.50" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-0-1", new String[] { "2.50" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-1-0", new String[] { "1.00" });
+        requestParams.put(Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-1-1", new String[] { "2.00" });
+
+        assertTrue(rubricDetails.extractQuestionDetails(requestParams, FeedbackQuestionType.RUBRIC));
+        assertTrue(rubricDetails.hasAssignedWeights());
+        List<String> errors = rubricDetails.validateQuestionDetails(dummySessionToken);
+        assertEquals(0, errors.size());
+    }
+
+    @Test
+    public void testGetRubricWeights_rubricWeightsForEachCellEmpty_legacyDataWillBeConvertedToNewFormat()
+            throws Exception {
+        // Create a dummy question to test the method.
+        String questionMetaDataString = "{\"rubricSubQuestions\":[\"This student has done a good job.\","
+                + "\"This student has tried his/her best.\"],"
+                + "\"questionText\":\"Please choose the best choice for the following sub-questions.\","
+                + "\"hasAssignedWeights\": true,"
+                + "\"rubricWeights\": [1.2, 1.84],"
+                + "\"numOfRubricChoices\":2,\"numOfRubricSubQuestions\":2,\"questionType\":\"RUBRIC\","
+                + "\"rubricChoices\":[\"Yes\",\"No\"],\"rubricDescriptions\":[[\"\",\"\"],"
+                + "[\"Most of the time\",\"Less than half the time\"]]}";
+
+        Text questionMetaData = new Text(questionMetaDataString);
+
+        List<FeedbackParticipantType> participants = new ArrayList<>();
+        participants.add(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+        participants.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionsDb fqdb = new FeedbackQuestionsDb();
+        FeedbackQuestionAttributes fqa = FeedbackQuestionAttributes.builder()
+                .withCourseId("testingCourse")
+                .withCreatorEmail("instructor@email.com")
+                .withFeedbackSessionName("testFeedbackSession")
+                .withGiverType(FeedbackParticipantType.INSTRUCTORS)
+                .withRecipientType(FeedbackParticipantType.SELF)
+                .withNumOfEntitiesToGiveFeedbackTo(1)
+                .withQuestionNumber(1)
+                .withQuestionType(FeedbackQuestionType.RUBRIC)
+                .withQuestionMetaData(questionMetaData)
+                .withShowGiverNameTo(new ArrayList<>(participants))
+                .withShowRecipientNameTo(new ArrayList<>(participants))
+                .withShowResponseTo(new ArrayList<>(participants))
+                .build();
+
+        FeedbackQuestion question = fqdb.createEntityWithoutExistenceCheck(fqa);
+
+        FeedbackQuestionAttributes attr = FeedbackQuestionAttributes.valueOf(question);
```

Comment:
The above two lines also don't need.

Sorry for my misleading review comments if any.

Reply:
Therefore, you can just extends `BaseTestCase`, not need to have `BaseAttributeTest`